tck/src/main/java/org/eclipse/microprofile/lra/tck/TckLRATypeTests.java

```diff
@@ -0,0 +1,349 @@
+
+/**
+ * <p>
+ * Tests that validate that the implementation of the {@link LRA.Type} attribute is correct
+ * </p>
+ *
+ * <p>
+ * For each of the type values there are 4 tests:
+ * <ul>
+ *     <li>start an LRA before invoking the resource method annotated with {@link LRA#end()} set to true</li>
+ *     <li>do not start an LRA before invoking the resource method annotated with {@link LRA#end()} set to true</li>
+ *     <li>start an LRA before invoking the resource method annotated with {@link LRA#end()} set to false</li>
+ *     <li>do not start an LRA before invoking the resource method annotated with {@link LRA#end()} set to false</li>
+ * </ul>
+ * </p>
+ *
+ * <p>
+ * Each test then validates {@link TckLRATypeTests#resourceRequest} that:
+ * <ul>
+ *     <li>the correct JAX-RS status was returned</li>
+ *     <li>the resource method executed with the correct LRA context</li>
+ *     <li>the LRA started by the resource method is closed if {@link LRA#end()} is true</li>
+ *     <li>the LRA started by the resource method is not closed if {@link LRA#end()} is false</li>
+ * </ul>
+ * </p>
+ */
+@RunWith(Arquillian.class)
+public class TckLRATypeTests {
+    private static final Logger LOGGER = Logger.getLogger(TckLRATypeTests.class.getName());
+
+    @Rule public TestName testName = new TestName();
+
+    @Inject
+    private LraTckConfigBean config;
+
+    private LRAClientOps lraClient;
+
+    private static Client tckSuiteClient;
+
+    private WebTarget tckSuiteTarget;
+
+    @Deployment(name = "lra-type-tck-tests", managed = true, testable = true)
+    public static WebArchive deploy() {
+        String archiveName = TckLRATypeTests.class.getSimpleName().toLowerCase();
+        return ShrinkWrap
+            .create(WebArchive.class, archiveName + ".war")
+            .addPackages(true, "org.eclipse.microprofile.lra.tck")
+            .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+    
+    @AfterClass
+    public static void afterClass() {
+        if(tckSuiteClient != null) {
+            tckSuiteClient.close();
+        }
+    }
+
+    @Before
+    public void before() {
+        LOGGER.info("Running test: " + testName.getMethodName());
+        setUpTestCase();
+
+        try {
+            tckSuiteTarget = tckSuiteClient.target(URI.create(new URL(config.tckSuiteBaseUrl()).toExternalForm()));
+            lraClient = new LRAClientOps(tckSuiteTarget);
+        } catch (MalformedURLException mfe) {
+            throw new IllegalStateException("Cannot create URL for the LRA TCK suite base url " + config.tckSuiteBaseUrl(), mfe);
+        }
+    }
+
+    private void setUpTestCase() {
+        tckSuiteClient = ClientBuilder.newClient();
+    }
+
+    // enum to indicate which checks to perform on the expected and actual active LRA after running a resource method
+    private enum MethodLRACheck {
+        NONE, NOT_PRESENT, EQUALS, NOT_EQUALS
+    }
+
+    @Test
+    public void requiredWithLRATest() {
```

Comment:
just a not to consider. When all those are tests not be more readable to name them just `requiredWithLRA`, `requiredWithoutLRA`

Reply:
+1